src/ImageManipulation.php

```diff
@@ -581,20 +581,25 @@ public function getIcon()
     }
 
     /**
+     * Cache of image backends
+     *
+     * @var array
+     */
+    protected static $image_backend_cache = [];
+
+    /**
      * Get Image_Backend instance for this image
      *
      * @return Image_Backend
      */
     public function getImageBackend()
     {
-        // Skip if non-image or file is broken
-        if (!$this->getIsImage() || !$this->exists()) {
-            return null;
+        // Pass to backend service factory
+        $backend = Injector::inst()->createWithArgs(Image_Backend::class, array($this));
+        if ($backend instanceof Image_Backend) {
+            return $backend;
```

Comment:
Is this to ensure that `InvalidBackend` doesn't make it through?

Also noticed `$this->image_backend_cache` hasn't been used yet, and may not be needed anymore :)